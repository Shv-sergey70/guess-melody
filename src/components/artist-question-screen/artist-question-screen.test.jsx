import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ArtistQuestionScreen from "./artist-question-screen";

configure({adapter: new Adapter()});

const question = {
  type: `artist`,
  song: {
    artist: `Artur Latte`,
    src: `test/track1.mp3`,
  },
  answers: [
    {
      picture: `test/track2.mp3`,
      artist: `Billie Eilish`
    },
    {
      picture: `test/track1.mp3`,
      artist: `Imagine Dragons`
    },
    {
      picture: `test/track3.mp3`,
      artist: `Artur Latte`
    }
  ]
};

test(`ArtistQuestionScreen correctly renders`, () => {
  const tree = shallow(
      <ArtistQuestionScreen
        question={question}
        screenIndex={3}
        onAnswer={jest.fn()} />
  );

  expect(tree).toMatchSnapshot();
});
