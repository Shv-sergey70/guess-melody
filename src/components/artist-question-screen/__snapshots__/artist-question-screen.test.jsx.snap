// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen correctly renders 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    >
      <audio
        id={0}
        src="test/track1.mp3"
      />
    </div>
  </div>
  <ArtistAnswersList
    onArtistSelect={[Function]}
    questionAnswers={
      Array [
        Object {
          "artist": "Billie Eilish",
          "picture": "test/track2.mp3",
        },
        Object {
          "artist": "Imagine Dragons",
          "picture": "test/track1.mp3",
        },
        Object {
          "artist": "Artur Latte",
          "picture": "test/track3.mp3",
        },
      ]
    }
    screenIndex={3}
  />
</section>
`;
